### PR TITLE
feat(speech): Adds speech V2 samples

### PR DESCRIPTION
**Pull request by telpirion**
This PR (1 of 3) adds the Speech V2 samples for Java.

Canonical samples are [here](https://github.com/googleapis/python-speech/pull/460).

### CODE REVIEW

File: speech/src/main/java/com/example/speech/CreateRecognizerV2.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      ExecutionException {
+    String projectId = "my-project-id";
+    String recognizerId = "my-recognizer";
+    
+    createRecognizerV2(projectId, recognizerId);
+  }
+
+  public static void createRecognizerV2(String projectId, String recognizerId) throws IOException,
+      InterruptedException, ExecutionException {
+    // Instantiate the SpeechClient
```
Comment (averikitsch) [inline speech/src/main/java/com/example/speech/CreateRecognizerV2.java:38]:
```suggestion
    // Initialize client that will be used to send requests. This client only needs to be created
    // once, and can be reused for multiple requests. After completing all of your requests, call
    // the "close" method on the client to safely clean up any remaining background resources.
```

File: speech/src/main/java/com/example/speech/CreateRecognizerV2.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    try (SpeechClient speechClient = SpeechClient.create()) {
+      String parent = String.format("projects/%s/locations/global", projectId);
+
+      // First, create a recognizer
```
Comment (averikitsch) [inline speech/src/main/java/com/example/speech/CreateRecognizerV2.java:42]: nit: comment not needed

File: speech/src/test/java/com/example/speech/CreateRecognizerV2Test.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    System.setOut(out);
+    projectId = ServiceOptions.getDefaultProjectId();
+
+    recognizerName = RecognizerName.format(projectId, "global", recognizerId);
+  }
+
+  @After
+  public void tearDown() throws IOException, ExecutionException, InterruptedException,
+      TimeoutException {
+    System.setOut(null);
```
Comment (averikitsch) [inline speech/src/test/java/com/example/speech/CreateRecognizerV2Test.java:61]: This should be reset to the original out, see example https://github.com/GoogleCloudPlatform/java-docs-samples/blob/c3aee37ee5fd593447b722faa6498fbaef45088f/datacatalog/snippets/src/test/java/com/example/datacatalog/ListEntryGroupsIT.java#L68

File: speech/src/test/java/com/example/speech/CreateRecognizerV2Test.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  private ByteArrayOutputStream bout;
+  private PrintStream out;
+
+  @Before
+  public void setUp() throws IOException, ExecutionException, InterruptedException,
+      TimeoutException {
+    bout = new ByteArrayOutputStream();
+    out = new PrintStream(bout);
+    System.setOut(out);
+    projectId = ServiceOptions.getDefaultProjectId();
```
Comment (averikitsch) [inline speech/src/test/java/com/example/speech/CreateRecognizerV2Test.java:53]: This is probably fine but we usually use env var "GOOGLE_CLOUD_PROJECT"

File: speech/src/test/java/com/example/speech/QuickstartSampleV2Test.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  @Before
+  public void setUp() {
+    bout = new ByteArrayOutputStream();
+    out = new PrintStream(bout);
+    System.setOut(out);
+    projectId = ServiceOptions.getDefaultProjectId();
+  }
+
+  @After
+  public void tearDown() {
```
Comment (averikitsch) [inline speech/src/test/java/com/example/speech/QuickstartSampleV2Test.java:50]: Please clean up the resources for this test.

**Review by averikitsch — CHANGES_REQUESTED** [review]

File: speech/src/main/java/com/example/speech/CreateRecognizerV2.java (hunk 2 above)
Comment (telpirion) [reply speech/src/main/java/com/example/speech/CreateRecognizerV2.java]: Removed.

File: speech/src/test/java/com/example/speech/CreateRecognizerV2Test.java (hunk 3 above)
Comment (telpirion) [reply speech/src/test/java/com/example/speech/CreateRecognizerV2Test.java]: Done.

File: speech/src/test/java/com/example/speech/CreateRecognizerV2Test.java (hunk 4 above)
Comment (telpirion) [reply speech/src/test/java/com/example/speech/CreateRecognizerV2Test.java]: Done.

File: speech/src/test/java/com/example/speech/QuickstartSampleV2Test.java (hunk 5 above)
Comment (telpirion) [reply speech/src/test/java/com/example/speech/QuickstartSampleV2Test.java]: Done.

File: speech/src/main/java/com/example/speech/CreateRecognizerV2.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+      ExecutionException {
+    String projectId = "my-project-id";
+    String recognizerId = "my-recognizer";
+    
+    createRecognizerV2(projectId, recognizerId);
+  }
+
+  public static void createRecognizerV2(String projectId, String recognizerId) throws IOException,
+      InterruptedException, ExecutionException {
+    // Initialize client that will be used to send requests. This client only needs to be created
```
Comment (averikitsch) [inline speech/src/main/java/com/example/speech/CreateRecognizerV2.java:38]: Please add to all samples

File: speech/src/test/java/com/example/speech/CreateRecognizerV2Test.java [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+  private String recognizerName;
+  private String projectId = System.getenv("GOOGLE_CLOUD_PROJECT");
+  private ByteArrayOutputStream bout;
+  private PrintStream out;
+
+  @Before
+  public void setUp() throws IOException, ExecutionException, InterruptedException,
+      TimeoutException {
+    bout = new ByteArrayOutputStream();
+    out = new PrintStream(bout);
```
Comment (averikitsch) [inline speech/src/test/java/com/example/speech/CreateRecognizerV2Test.java:53]: My last comment was not clear. We need to save and restore to the original stdout
```suggestion
    out = new PrintStream(bout);
    originalPrintStream = System.out;
```

File: speech/src/test/java/com/example/speech/CreateRecognizerV2Test.java [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+    System.setOut(out);
+
+    recognizerName = RecognizerName.format(projectId, "global", recognizerId);
+  }
+
+  @After
+  public void tearDown() throws IOException, ExecutionException, InterruptedException,
+      TimeoutException {
+    System.setOut(out);
```
Comment (averikitsch) [inline speech/src/test/java/com/example/speech/CreateRecognizerV2Test.java:62]: Resetting to original stdout
```suggestion
    System.setOut(originalPrintStream);
```

**Review by averikitsch — APPROVED** [review]
Approving, but please make sure all samples are updated (see comments)

**Comment by averikitsch** [comment]
Tests are not running, I believe they need to be updated with IT.java suffix not Test.java:
```
#
# speech
#

[0.001s][warning][os,container] Duplicate cpuset controllers detected. Picking /sys/fs/cgroup/cpuset, skipping /tmpfs/docker/overlay2/2a80f6b970fdd50649efd99b3c9ec55fce1046f7c2f6a67c4fb497001642b84c/merged/sys/fs/cgroup/cpuset.

 Testing completed.




#
# Summary
#

SUCCESS: 1, FAILURE: 0, SKIPPED: 0, ERROR: 0, 
speech................................................................[ SUCCESS]
```

File: speech/src/main/java/com/example/speech/CreateRecognizerV2.java (hunk 6 above)
Comment (telpirion) [reply speech/src/main/java/com/example/speech/CreateRecognizerV2.java:39]: Okay! Done.

File: speech/src/test/java/com/example/speech/CreateRecognizerV2Test.java (hunk 7 above)
Comment (telpirion) [reply speech/src/test/java/com/example/speech/CreateRecognizerV2Test.java]: Done.

File: speech/src/test/java/com/example/speech/CreateRecognizerV2Test.java (hunk 8 above)
Comment (telpirion) [reply speech/src/test/java/com/example/speech/CreateRecognizerV2Test.java]: Done.

**Comment by telpirion** [comment]
Test renamed; hopefully they'll run now!

**Pull request merged**